File: src/lib/pjecalc/core/dominio/calculo/relatorio/justificativa-jr-adapter.ts
/**
 * PJe-Calc v2.15.1 — JustificativaJRAdapter (abstract)
 * Porte TS-adaptado de:
 *   br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.JustificativaJRAdapter
 *
 * Ref Java: pjecalc-fonte/.../relatorio/JustificativaJRAdapter.java
 *
 * Seção de justificativas (notas explicativas) do relatório do cálculo.
 */
import { JRAdapter, JRAdapterDataSource } from './jr-adapter';

export abstract class JustificativaJRAdapter extends JRAdapter {
  abstract getJustificativas(): JRAdapterDataSource<JustificativaItemAdapter>;
}

export abstract class JustificativaItemAdapter extends JRAdapter {
  abstract getTitulo(): string;
  abstract getTexto(): string;
}

File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/calculo/relatorio/JustificativaJRAdapter.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio;

import br.jus.trt8.pjecalc.base.comum.relatorio.JRAdapter;
import br.jus.trt8.pjecalc.base.comum.relatorio.JRAdapterDataSource;

public abstract class JustificativaJRAdapter
extends JRAdapter {
    public abstract JRAdapterDataSource<JustificativaOcorrenciaAdapter> getOcorrencias();

    public abstract class JustificativaOcorrenciaAdapter
    extends JRAdapter {
        public abstract Integer getNumero();

        public abstract String getTextoReal();
    }
}

